src/widget/wnumberrate.cpp

```diff
@@ -32,8 +31,11 @@ void WNumberRate::setup(const QDomNode& node, const SkinContext& context) {
 void WNumberRate::setValue(double dValue) {
     double vsign = dValue - 1;
 
-    char sign = '+';
-    if (vsign < -0.00000001) {
+    QString sign(' ');
+    if(vsign > 0.0001) {
+        sign = '+';
+    }
+    if(vsign < 0.0001) {
```

Comment:
- This condition is wrong
- The magic constant 0.0001 depends on m_iNoDigits and should not be hardcoded
- Please fix the code formatting

Reply:
If this delta is desired then the transformation into the final percent value has to be done beforehand and not inline as it is done now (see below).

Reply:
@uklotzde can you make a commit suggestion? It seems there is misunderstanding here with the condition and formatting, am not sure what you mean exactly

Reply:
@uklotzde Do you mean something like this?

```cpp
double digitFactor = pow(10, iNoDigits);
// Calculate percentage rounded to the number of digits specified by iNoDigits
double percentage = round((dValue - 1) * 100.0 * digitFactor) / digitFactor;

QString sign(' ');
if(percentage > 0) {
    sign = '+';
}
if(percentage < 0) {
    sign = '-';
}

setText(QString(m_skinText).append(sign).append("%1").arg(fabs(percentage), 'f', iNoDigits);
```

Reply:
- digitFactor/percentage :+1: 
- `sign` should be implemented as a function `inline QChar sign(double value)` in an anonymous namespace using `if`/`else if`/`else` for the 3 different return paths
- `if (` is missing a separating space character
- finally all commits should be squashed to reduce the noise

Reply:
`setText(m_skinText + sign(percentage) + QString::number(fabs(percentage), 'f', m_iNoDigits));`

Reply:
This is how readable and maintainable code should look like.

Reply:
I think this should be better now